lib/storage/s3.js

```diff
@@ -132,8 +133,16 @@ module.exports = function() {
           inputStream.pipe(gunzip);
```

Comment:
This existing code takes care of gunzip on the fly when the content encoding is present.